src/DotVVM.Framework/Hosting/DotvvmRequestContextExtensions.cs

```diff
@@ -83,14 +85,37 @@ public static void RedirectToUrl(this IDotvvmRequestContext context, string url,
         /// <summary>
         /// Returns the redirect response and interrupts the execution of current request.
         /// </summary>
-        public static void RedirectToRoute(this IDotvvmRequestContext context, string routeName, object newRouteValues = null, bool replaceInHistory = false, bool allowSpaRedirect = true, string urlSuffix = null)
+        public static void RedirectToRoute(this IDotvvmRequestContext context, string routeName, object newRouteValues = null, bool replaceInHistory = false, bool allowSpaRedirect = true, string urlSuffix = null, object queryStringParameters = null)
```

Comment:
I'd rather use a shorter name because this parameter is going to be used as a named one, what about `query`? I think it's quite clear what it means the context of redirection.